common/src/Utilities/PString.cpp

```diff
@@ -0,0 +1,81 @@
+#include "PrecompiledHeader.h"
+#include "PString.h"
+#include <codecvt>
+#include <wchar.h>
+
+const PString& PString::operator=(const PString& rhs)
+{
+	string = rhs.string;
+	return *this;
+}
+
+PString& PString::operator=(PString&& rhs)
+{
+	string = std::move(rhs.string);
+	return *this;
+}
+
+#ifdef _WIN32
+// Non default
+PString::PString(const std::wstring& utf16_string)
+{
+	const int size = WideCharToMultiByte(CP_UTF8, 0, utf16_string.c_str(), utf16_string.size(), nullptr, 0, nullptr, nullptr);
+	std::string converted_string(size, 0);
+	WideCharToMultiByte(CP_UTF8, 0, utf16_string.c_str(), utf16_string.size(), converted_string.data(), converted_string.size(), nullptr, nullptr);
+	string = converted_string;
+}
+
+PString::operator std::wstring()
+{
+	int size = MultiByteToWideChar(CP_UTF8, 0, string.c_str(), -1, nullptr, 0);
```

Comment:
You have access to `string.size()` here, no need to rely on the null terminator.